packages/node_modules/@webex/common/src/patterns.js

```diff
@@ -11,12 +11,23 @@
  */
 export default {
   /**
-   * Matches an email address by requiring an @ and excluding spaces
-   * @todo add a better email address matcher
+   * Matches an email address by requiring an @ and excluding spaces. This is
+   * the legacy regex for methods that explicitly look for a string containing
+   * only the email, and nothing else.
+   *
    * @type {RegExp}
    */
   email: /^[^\s]+?@[^\s]+?$/,
```

Comment:
Is there a reason to keep the old one? Would the new RegEx not detect an email if it's not on its own?

Reply:
Legacy conditions:
```js
const str1 = 'email@website.com';
const str2 = 'this is an email: email@website.com';

email.test(str1) // true
email.test(str2) // false

emailInString.test(str1) // true
emailInString.test(str2) // true
```

In the case that our plugins are attempting to determine if a value is explicitly an email, the original regex is needed.

In the case in which we are looking for an email within a varying string, the new regex is needed.

Usecase: Authorization [validating if a key:value is an email for authentication].

Reply:
From your code it looks like the new regex works for the old scenario, too. I want to avoid unnecessary code

Reply:
Code snippet may have been misleading due to the requisite of a negative, see the following for a specific use case:

```js
// Validates if the input provided is a valid email
function checkIfLoginEmailValid(providedEmail) {
  // providedEmail = "hello world email@website.com" for example

  return email.test(providedEmail); // this returns false, which is correct
  return emailInString.test(providedEmail): // returns true, which is incorrect
}
```

Let me know if this makes sense.